### PR TITLE
refactor(*): isolate all electron specific code in platform specific files

### PR DESCRIPTION
**Pull request by ramfox**
This will allow us to use the contents of the `app` to build a webapp without needing electron.

relates to #663

### CODE REVIEW

File: app/components/platformSpecific/ContextMenuArea.web.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Context menus only make sense in context of the electron app right now
+ * as all the actions that one can take are dependent on the dataset being
+ * either in your namespace and/or should be actions that happen only if
+ * you are working locally
+ */
+import * as React from 'react'
```
Comment (b5) [inline app/components/platformSpecific/ContextMenuArea.web.tsx:7]: This syntax is now a little antiquated. Preferred import method is just:

```tsx
import React from 'react`
```

File: app/components/nav/NavSidebar.tsx [hunk 2]
```diff
@@ -159,7 +160,11 @@ export const NavbarComponent: React.FunctionComponent<NavbarProps> = (props: Nav
           <NavbarItem
             icon={faComment}
             tooltip={'Need help? Ask questions<br/> in our Discord channel'}
-            onClick={() => { shell.openExternal(DISCORD_URL) }}
+            // when we have a function for `openInExternalWindow` we don't need
+            // a link to get us there, and vise-versa
+            onClick={() => { openInExternalWindow && openInExternalWindow(DISCORD_URL) }}
+            link={!openInExternalWindow && DISCORD_URL}
+            externalLink
```
Comment (b5) [inline app/components/nav/NavSidebar.tsx:167]: could refactor this as either a one-off `openDiscord` function, or a switch on a `__BUILD__.TARGET_PLATFORM === `web` && ...` compile-time flag

**Review by b5 — APPROVED** [review]
A few nits, then LGTM!

File: app/components/App.tsx [hunk 3]
```diff
@@ -2,10 +2,18 @@ import * as React from 'react'
 import { Action } from 'redux'
 import { CSSTransition } from 'react-transition-group'
 import { ConnectedRouter, push } from 'connected-react-router'
-import { ipcRenderer, remote } from 'electron'
 import fs from 'fs'
```
[the remark this answers is not in the document]
Comment (ramfox) [reply app/components/App.tsx:5]: Hey, I'm a little confused about what you are referencing? Are you saying that 'fs' is a node dependency? And if so, why is that an issue?

**Pull request merged**